src/components/radio-button/radiobutton.test.js

```diff
@@ -0,0 +1,565 @@
+    (position) => {
+      CypressMountWithProviders(
+        <RadiobuttonComponent
+          labelHelp="Tooltip info"
+          tooltipPosition={position}
+        />
+      );
+      radiobuttonIcon().trigger("mouseover");
+      tooltipPreview()
+        .should("have.text", "Tooltip info")
+        .should("have.attr", "data-placement", `${position}`);
+    }
+  );
+
+  it("should render Radiobutton component with fill color set to black by default", () => {
+    CypressMountWithProviders(<RadiobuttonComponent checked />);
+    radiobuttonRole()
+      .should("be.checked")
+      .should("have.css", "color", "rgb(0, 0, 0)");
+  });
+
+  describe("should render Radiobutton component and check events", () => {
+    let callback;
+
+    beforeEach(() => {
+      callback = cy.stub();
+    });
+
+    it("should call onBlur callback when a blur event is triggered", () => {
+      CypressMountWithProviders(<RadiobuttonComponent onBlur={callback} />);
+
+      radiobuttonRole()
+        .focus()
+        .blur()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call onChange callback when a check event is triggered", () => {
+      CypressMountWithProviders(<RadiobuttonComponent onChange={callback} />);
+
+      radiobuttonRole()
+        .check()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call onFocus callback when a focus event is triggered", () => {
+      CypressMountWithProviders(<RadiobuttonComponent onFocus={callback} />);
+
+      radiobuttonRole()
+        .focus()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call onClick callback when a click event is triggered", () => {
+      CypressMountWithProviders(<RadiobuttonComponent onClick={callback} />);
+
+      radiobuttonRole()
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    describe("Testing RadiobuttonGroup component", () => {
+      it.each(specialCharacters)(
+        "should render RadiobuttonGroup component with %s as legend",
+        (legendValue) => {
+          CypressMountWithProviders(
+            <RadiobuttonGroupComponent legend={legendValue} />
+          );
+          radiobuttonGroupLegend().should("have.text", legendValue);
+        }
+      );
+
+      it("should render RadiobuttonGroup component with error", () => {
+        CypressMountWithProviders(<RadiobuttonGroupComponent error />);
+
+        radiobuttonSvg().should(
+          "have.css",
+          "border-bottom-color",
+          "rgb(205, 56, 75)"
```

Comment:
should we probably create a const for each color and use them in the `validation` tests?

Reply:
Added constants for the validation colors